relay-server/src/utils/sleep_handle.rs

```diff
@@ -0,0 +1,35 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::Poll;
+use std::time::Duration;
+
+pub struct SleepHandle(Option<Pin<Box<tokio::time::Sleep>>>);
```

Comment:
we should add a doc comment to this though, and all the pub methods.  Should describe it is so that you can have a future that either is always pending or wakes up after some time.

Also that setting it while it is being polled will not work (I think), which is a very important caveat.

Reply:
oh wait, i'm talking nonsense again. poll is &mut self so that's already enforced by the typesystem